x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTask.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.async;
+
+import java.util.Map;
+
+/**
+ * A task that supports asynchronous execution and safe temporary storage of results
```

Comment:
given that it's not part of the interface, maybe storing results should not be mentioned?